enterprise/server/remote_execution/uffd/uffd.go

```diff
@@ -177,8 +188,16 @@ func (h *Handler) handle(ctx context.Context, memoryStore *blockio.COWStore) err
 	}
 
 	for {
+		select {
+		case <-h.quitChan:
+			return nil
+		default:
+		}
+
 		// Poll UFFD for messages
-		_, pollErr := unix.Poll(pollFDs, -1)
+		// Poll blocks until a message is ready to be read. A timeout is necessary so that if the handler receives
+		// a shutdown message on quitChan while Poll is waiting, Poll will exit and restart the loop to check quitChan
+		_, pollErr := unix.Poll(pollFDs, 500 /* timeout in ms */)
```

Comment:
This timeout means that `Remove()` can take 500ms which seems a bit too long if we can avoid it.

It looks like a pattern for "canceling" a poll is to create a file descriptor that you control (in golang you can use `os.Pipe()` to get read/write ends of a pipe and call `Fd()` to get the file descriptor. Then you can add the output end of the pipe to the `pollFDs` list, and to cancel, write something to the input end of the pipe.

See https://stackoverflow.com/a/4049050

Reply:
Ah very clever! Updated